include/irr/asset/filters/CSummedAreaTableImageFilter.h

```diff
@@ -0,0 +1,272 @@
+// Copyright (C) 2020 - AnastaZIuk
+// This file is part of the "IrrlichtBAW" engine.
+// For conditions of distribution and use, see copyright notice in irrlicht.h
+
+#ifndef __IRR_C_SUMMED_AREA_TABLE_IMAGE_FILTER_H_INCLUDED__
+#define __IRR_C_SUMMED_AREA_TABLE_IMAGE_FILTER_H_INCLUDED__
+
+#include "irr/core/core.h"
+
+#include <type_traits>
+#include <functional>
+
+#include "irr/asset/filters/CMatchedSizeInOutImageFilterCommon.h"
+#include "CConvertFormatImageFilter.h"
+
+namespace irr
+{
+namespace asset
+{
+
+template<bool ExclusiveMode>
+class CSummedAreaTableImageFilterBase
+{
+	public:
+		class CSummStateBase
+		{
+			public:
+				
+				uint8_t*	scratchMemory = nullptr;										//!< memory covering all regions used for temporary filling within computation of sum values
+				size_t	scratchMemoryByteSize = {};											//!< required byte size for entire scratch memory
+
+				static inline size_t getRequiredScratchByteSize(const ICPUImage* inputImage)
+				{
+					constexpr auto decodeByteSize = 8u;
```

Comment:
`sizeof(double)` makes it more readable, otherwise people are scratching their heads like "why 8" ?

Reply:
you're right, to fix